packages/grafana-data/src/field/fieldState.ts

```diff
@@ -38,14 +38,17 @@ export function getFrameDisplayName(frame: DataFrame, index?: number) {
 
 export function getFieldDisplayName(field: Field, frame?: DataFrame, allFrames?: DataFrame[]): string {
   const existingTitle = field.state?.displayName;
+  const canUseCachedDisplayName =
+    !Boolean(field.state?.multipleFrames) || (Boolean(field.state?.multipleFrames) && Boolean(allFrames));
```

Comment:
Is not the same as `Boolean(field.state?.multipleFrames) ===  Boolean(allFrames && allFrames.length > 1)`

Reply:
The expression above breaks some tests. 
Also, maybe I don't understand correctly, but why should we allow the use of the cached display name if both multipleFrames and allFrames are either truthy/falsy?